You must use Pandas to answer the following questions:

How many people of each race are represented in this dataset? This should be a Pandas series with race names as the index labels. (race column) OKAY

What is the average age of men? OKAY

What is the percentage of people who have a Bachelor's degree? OKAY

What percentage of people with advanced education (Bachelors, Masters, or Doctorate) make more than 50K? OKAY

What percentage of people without advanced education make more than 50K? OKAY

What is the minimum number of hours a person works per week? OKAY

What percentage of the people who work the minimum number of hours per week have a salary of more than 50K? OKAY

What country has the highest percentage of people that earn >50K and what is that percentage? OKAY

Identify the most popular occupation for those who earn >50K in India. OKAY

Use the starter code in the file demographic_data_analyzer. Update the code so all variables set to "None"
are set to the appropriate calculation or code. Round all decimals to the nearest tenth.

In [1]:
import numpy as np
import pandas as pd
import math
import operator

In [2]:
file= pd.read_csv('adult.data.csv')

In [3]:
df = pd.DataFrame(file)
#df.info()
df.head(5)

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

***1- How many people of each race are represented in this dataset? This should be a Pandas series with race names as the index labels. (race column)***

In [7]:
raza=df['race']
raza= np.array(raza)

#White race
white= raza[raza=='White']
white_tot=len(white)

#Black race
black= raza[raza=='Black']
black_tot=len(black)

#Otras se definio para ir explorando el DF y ver que razas habian. Son 5 en total
other= raza[~ ((raza=='White') | (raza=='Black') | (raza== 'Asian-Pac-Islander')
            | (raza=='Amer-Indian-Eskimo'))]
other_tot=len(other)

As_Pac_Islander= raza[raza =='Asian-Pac-Islander']
As_Pac_Islander_tot= len(As_Pac_Islander)

Amer_Indian_Eskimo= raza[raza== 'Amer-Indian-Eskimo']
Amer_Indian_Eskimo_tot= len(Amer_Indian_Eskimo)

a=white_tot
b=black_tot
c=As_Pac_Islander_tot
d=Amer_Indian_Eskimo_tot
e=other_tot
Final=pd.Series([a,b,c,d,e])
Final.index= ['White', 'Black', 'Asian-Pac-Islander', 'Amer-Indian-Eskimo', 'Other']
print(Final, type(Final))



White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
dtype: int64 <class 'pandas.core.series.Series'>


In [8]:
#REPETICION USANDO DICCIONARIOS: LA POSTA
raza=df['race']
raza= np.array(raza)

#Creation of a dictionary to count the people of each race
race_dict=dict()
for races in raza:
    race_dict[races] = race_dict.get(races,0) + 1

#Create 2 lists to save the total number and race
race_list= []
race_ind= []
for key, value in race_dict.items():
    race_list.append(value)
    race_ind.append(key)
    
    
#Convert the previous lists into pd.Series 
race= pd.Series(race_list, index= race_ind)
print(race)


#print (race_list, (race_ind))

White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
dtype: int64


***2-Respuesta a la pregunta sobre la edad promedio de los hombres:***

In [5]:
#La porcion de la gente que es de sexo masculino está dada por:
A=df.loc[df['sex'] == 'Male']

#Y la edad promedio es:
round(A['age'].mean(),1)

39.4

***3- Porcentaje de gente con "bachelors degree" en educacion:***

In [9]:
Bachelors_Count= sum(df['education']=='Bachelors')
Total_Count= sum(df['education'] != 'Bachelors') + Bachelors_Count
#print(Bachelors_Count)
#print(Total_Count)
round((Bachelors_Count/Total_Count * 100),1)

16.4

El porcentaje de gente con Bachelors es de 16.4%

***4-What percentage of people with advanced education (Bachelors, Masters, or Doctorate) make more than 50K?***

In [9]:
Rich=df[(df['education'] == 'Bachelors') | (df['education'] == 'Masters') | (df['education'] == 'Doctorate')]
#Rich

# Total de la gente con educacion avanzada que gana arriba de 50 K
more50K=sum(Rich['salary'] == '>50K')
# Total de gente con educacion avanzada
total_adv=len(Rich['salary'])
# Porcentaje de gente que gana arriba de 50K entre los que tienen educacion avanzada
percentage_adv= round(more50K/total_adv * 100,1) 
percentage_adv
#print(total_adv)

46.5

***5-What percentage of people without advanced education make more than 50K?***

In [10]:
#Creo un dataframe con la gente que no tiene educacion avanzada:
Rich_normal= df[~( (df['education'] == 'Bachelors') | 
                   (df ['education'] == 'Masters')  | 
                   (df['education'] == 'Doctorate') )]

percentage_normal= sum(Rich_normal['salary'] == '>50K') / len(Rich_normal['salary']) * 100

#Redondeo:
percentage_normal= round(percentage_normal,1)
print(percentage_normal)

17.4


***6-What is the minimum number of hours a person works per week?***

In [11]:
#print(df.head(4))
min_hours_week = df['hours-per-week'].min()
print(min_hours_week)

1


***7-What percentage of the people who work the minimum number of hours per week have a salary of more than 50K?***


In [12]:
# Creo un DataFrame con la gente que trabaja 1 hora
X=df[df['hours-per-week'] == min_hours_week]

#La cantidad de gente que cumple el criterio(horas=1 y salario >50K) es:
a= sum ( X['salary'] == '>50K')

#El total de la poblacion que trabaja 1 hora por semana:
b= len(X)


perc_very_rich= round(a/b * 100, 1)
print(perc_very_rich)


10.0


***8-What country has the highest percentage of people that earn >50K and what is that percentage? OKAY!***

In [113]:
#USANDO UN DICCIONARIO EN VEZ DE UNA LISTA
#What country has the highest percentage of people that earn >50K and what is that percentage?

#Primero trabajo para ver cuantos paises distintos hay para 'native-country'
Paises= df['native-country']
print(type(Paises))

#En el siguiente diccionario aparecen los pares paises: apariciones totales
paises_dict= dict()
for country in Paises:
    paises_dict[country] = paises_dict.get (country, 0) + 1


#Seteo el mejor caso:
optimal_country= None
optimal_percentage = 0

dict_ingresos= dict() #Aca voy a guardar el porcentaje de gente con ingresos >50K

for pais,value in paises_dict.items():
    aux_DF= df.loc[df ['native-country'] == pais ]
    Total=value
    
    #La porcion del DF aux con el pais dado que tiene ingresos >50K:
    X = aux_DF[aux_DF['salary'] == '>50K']
    mayor50K= len(X)
    
    porcentaje= round( (mayor50K/Total * 100) , 1)
    dict_ingresos [pais]= porcentaje
        
    if porcentaje > optimal_percentage:
        optimal_percentage= porcentaje
        optimal_country= pais
        print(optimal_percentage, pais)
#print(dict_ingresos['Iran'])
    

<class 'pandas.core.series.Series'>
24.6 United-States
26.3 Cuba
40.0 India
41.9 Iran


***9-Identify the most popular occupation for those who earn >50K in India.***

In [73]:
India= df.loc[df['native-country']== 'India']
IndiaMore50K= India.loc[India['salary'] == ">50K"]

occupations= IndiaMore50K['occupation']

#Create a dictionary to get the number of people in each occupation
jobs_dict= dict()
for jobs in occupations:
    jobs_dict[jobs]=jobs_dict.get(jobs,0) + 1
    
#The dictionary is sorted according to number of workers in each ocupation (from higher to lower)
# A LIST IS CREATED! jobs_sorted is a list
jobs_sorted = sorted (jobs_dict.items(), key= operator.itemgetter(1), reverse= True)


Most_Pop_Job=(jobs_sorted[0][0])
print(Most_Pop_Job)
#print(jobs_sorted, type(jobs_sorted))


Prof-specialty
